code/TagField.php

```diff
@@ -173,4 +273,90 @@ public function saveInto(DataObjectInterface $record) {
 			$record->$name = implode(',', $values);
 		}
 	}
+
+	/**
+	 * Returns a JSON string of tags, for ajax-based search.
+	 *
+	 * @param SS_HTTPRequest $request
+	 *
+	 * @return string
+	 */
+	public function suggest(SS_HTTPRequest $request) {
+		if($this->recordClass === null) {
+			return '[]';
+		}
+
+		/**
+		 * @var DataObject $object
+		 */
+		$object = singleton($this->recordClass);
+
+		$term = Convert::raw2sql($request->getVar('term'));
+
+		$tags = array();
+
+		if($object->hasMethod($this->name)) {
+			$tags = $this->getObjectTags($this->recordClass, $object, $term);
+		} elseif($object->hasField($this->name)) {
+			$tags = $this->getStringTags($this->recordClass, $object, $term);
+		}
+
+		return Convert::raw2json($tags);
```

Comment:
Is it worth setting the content-type here? You can return a SS_HTTPResponse object instead with the necessary headers instead of a string return type.